src/main/java/pl/tomaszdziurko/jvm_bloggers/view/login/LoginPage.java

```diff
@@ -34,21 +39,26 @@
     @SpringBean
     private BruteForceLoginAttackDetector bruteForceLoginAttackDetector;
 
+    @SpringBean
+    private BruteForceAttackEventStreamManager bruteForceAttackEventStreamManager;
+
     public LoginPage() {
         StatelessForm<LoginPage> loginForm = new StatelessForm<LoginPage>(LOGIN_FORM_ID, new CompoundPropertyModel<>(this)) {
             @Override
             protected void onSubmit() {
-                String clientAddress = getClientAddress();
-                boolean bruteForceAttackDetected = bruteForceLoginAttackDetector.isItBruteForceAttack(clientAddress);
+                final String clientAddress = getClientAddress();
+                final boolean bruteForceAttackDetected = bruteForceLoginAttackDetector.isItBruteForceAttack(clientAddress);
                 if (bruteForceAttackDetected) {
-                    error("Incorrect login or password");
+                    error("Incorrect login or password [BruteForce attack was detected]");
+                    bruteForceAttackEventStreamManager.build(clientAddress)
+                            .publish(BruteForceAttackEvent.builder().ipAddress(clientAddress).build());
```

Comment:
It's not clear what is happening here - consider extracting a well named self describing method for the code:
```
 bruteForceAttackEventStreamManager.build(clientAddress)
        .publish(BruteForceAttackEvent.builder().ipAddress(clientAddress).build());
```